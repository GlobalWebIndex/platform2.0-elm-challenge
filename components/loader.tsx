export function Loader() {
  return <div className="loader fixed left-0 right-0 top-0 z-50 h-1"></div>
}
