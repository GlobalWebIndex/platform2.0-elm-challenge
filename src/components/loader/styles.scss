.loader {
  position: absolute;
  top: 30%;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 10;
  margin: 5rem auto;
  text-align: center;

  svg {
    height: 8rem;
    width: 8rem;
    fill: rgba(0, 0, 0, 0.5);
    transform-origin: 44% 50%;
    animation: rotate 1.5s infinite linear;
  }
}

@keyframes rotate {
  0% {
    transform: rotate(0);
  }
  100% {
    transform: rotate(360deg);
  }
}
