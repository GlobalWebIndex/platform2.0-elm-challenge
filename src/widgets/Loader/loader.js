import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CircularProgress from '@mui/material/CircularProgress';
import Box from '@mui/material/Box';
import { loadingSelector } from './selectors';

const Loader = ({ loading }) => (
  loading ? (
  <Box sx={{ display: 'flex' }}>
    <CircularProgress />
  </Box>
  ) : null
);

Loader.propTypes = {
  loading: PropTypes.bool,
};

const mapStateToProps = state => ({
  loading: loadingSelector(state)
});

export default connect(mapStateToProps)(Loader);