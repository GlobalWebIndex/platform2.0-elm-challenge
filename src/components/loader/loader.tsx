import React from 'react';
import Icons from '../../assets/svg/icons.svg';
import styles from './styles.scss';

export const Loader = (): JSX.Element => (
  <div className={styles.loader}>
    <svg>
      <use href={`${Icons}#icon-cw`}></use>
    </svg>
  </div>
);
