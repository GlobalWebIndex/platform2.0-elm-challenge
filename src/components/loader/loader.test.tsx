import React from 'react';
import {render} from '@testing-library/react';
import {Loader} from './loader';

describe('<Loader />', () => {
  it('should render loader', () => {
    const {container} = render(<Loader />);

    expect(container).toMatchSnapshot();
  });
});
