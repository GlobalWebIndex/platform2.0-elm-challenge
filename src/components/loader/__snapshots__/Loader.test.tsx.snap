// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render loader 1`] = `
<div>
  <div
    class="loader"
  >
    <svg>
      <use
        href="function SvgIcons(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgIcons.name})
  );
}#icon-cw"
      />
    </svg>
  </div>
</div>
`;
